velox/functions/sparksql/registration/RegisterArray.cpp

```diff
@@ -109,6 +109,27 @@ inline void registerArrayRemoveFunctions(const std::string& prefix) {
       Varchar>({prefix + "array_remove"});
 }
 
+template <typename T>
+inline void registerArrayUnionFunction(const std::string& prefix) {
+  registerFunction<ArrayUnionFunction, Array<T>, Array<T>, Array<T>>(
+      {prefix + "array_union"});
+}
+
+inline void registerArrayUnionFunctions(const std::string& prefix) {
+  registerArrayUnionFunction<int8_t>(prefix);
+  registerArrayUnionFunction<int16_t>(prefix);
+  registerArrayUnionFunction<int32_t>(prefix);
+  registerArrayUnionFunction<int64_t>(prefix);
+  registerArrayUnionFunction<int128_t>(prefix);
+  registerArrayUnionFunction<float>(prefix);
+  registerArrayUnionFunction<double>(prefix);
+  registerArrayUnionFunction<bool>(prefix);
+  registerArrayUnionFunction<Timestamp>(prefix);
+  registerArrayUnionFunction<Date>(prefix);
+  registerArrayUnionFunction<Varbinary>(prefix);
+  registerArrayUnionFunction<Generic<T1>>(prefix);
```

Comment:
Do we need the registration for Varchar?

Reply:
Good catch.
It should be noted that Generic<T1> ensures all types are registered, and individual registration is only for performance optimization.